### DISCUSSION TITLE
Create Csla.Mvvm, move ViewModelBase<T> & other non-UI specific classes to it

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#1765] [Ideas]
We're using Csla.XamarinForms & Csla.Wpf, both of which define ViewModelBase<T>.  We're trying to keep our VMs UI agnostic, as we'd like to move our WinForms app to WPF (or possibly UWP), and unify some use cases between our Mobile app, which is XF & desktop app.  

It seems like ViewModelBase<T> & possibly some other code in Csla.Xaml.Shared could really just be in a non-UI specific Csla.Mvvm library.  The advantage is that our NS VM library code wouldn't implicitly pull in a dependency on XF.  We're getting around this for now by having a separate .Net framework VM library with the VM code actually in a shared project, but it would be great to reduce things to a single .Net standard base.

Csla.XF could then continue to have things that actually depend on the UI like the PropertyInfo control.

This is similar to how MvvmLight has been splitting things up, with the Mvvm parts separated from the things that are really view related.

**Comment by rockfordlhotka** [comment]
I agree, and built [Bxf](https://github.com/MarimerLLC/Bxf) years ago to support some of what you are talking about.

Then I decided I didn't want to be sole maintainer of yet another mvvm framework, especially given the great work being done in MvvmLight and others. So I really haven't touched Bxf for a long time.

That said, I am open to the idea of a `Csla.Mvvm` package, as long as the scope isn't to be a full MVVM framework, but rather to provide implementation neutral support to make it easier for folks to use CSLA along with things like MvvmLight.

**Reply by rockfordlhotka** [reply]
Actually _moving_ types would be a breaking change of course. The next window for breaking changes is CSLA 6. My current timeframe for CSLA 6 corresponds with the release of .NET 5 this November, so time is relatively short. Though I'm not 100% certain that CSLA 6 does need to come out concurrent with .NET 5, because the scope of .NET 5 has shrunk thanks the pandemic, and some of the most pressing changes to CSLA are no longer coupled to the .NET 5 release.

**Reply by ajohnstone-ks** [reply]
No, I wasn't hoping for a full Mvvm framework; just some place for some of the helper stuff to do Mvvm with Csla.  The ViewModelBase is very useful in this regard and many of our VMs are built on top of it for edit use cases.  Of course moving to Wpf/Uwp will continue to be slow going for us regardless, so Csla 6 or 7 would be fine timing for us. :D

**Reply by rockfordlhotka** [reply]
I am very open to contributions - design and/or code 😄

**Reply by ajohnstone-ks** [reply]
I had a feeling that was coming 😆 I'll see when I can give it a whirl.

**Reply by rockfordlhotka** [reply]
Hopefully I captured your thoughts in this backlog issue: https://github.com/MarimerLLC/csla/issues/1766
